packages/build/src/plugins_core/deploy/buildbot_client.js

```diff
@@ -0,0 +1,71 @@
+const net = require('net')
+const { promisify } = require('util')
+
+const pEvent = require('p-event')
+
+const { addErrorInfo } = require('../../error/info')
+
+const BUILDBOT_CLIENT_TIMEOUT_PERIOD = 60 * 1000
+
+const createBuildbotClient = function(buildbotServerSocket) {
+  return net.createConnection(buildbotServerSocket)
+}
+
+const connectBuildbotClient = async function(client) {
+  try {
+    client.setTimeout(BUILDBOT_CLIENT_TIMEOUT_PERIOD, () => {
+      client.end()
+      const err = new Error('Buildbot client timed out')
```

Comment:
Throwing an error would emit an [`uncaughtException` event](uncaughtException) on `process`, resulting in `@netlify/build` to abort.
Using `client.destroy(new Error('...'))` instead would not: instead, it would emit an [`error` event](https://nodejs.org/api/net.html#net_event_error) on the TCP `client`. `p-event` [throws/rejects](https://github.com/sindresorhus/p-event#rejectionevents) on `error` events by default, which means `await pEvent(client, 'connect')` would throw, propagating the error to the `catch` block instead of aborting the process.